py/dynruntime.mk

```diff
@@ -124,6 +124,10 @@ else
 $(error architecture '$(ARCH)' not supported)
 endif
 
+ifneq ($(shell ldd --version 2>&1 | head -n 1 | grep -i musl),)
```

Comment:
Wouldn't something like `$(lastword $(subst -, ,$(shell $(CROSS)gcc -dumpversion))` (not tested but it should work) be more accurate rather than looking at the default dynamic linker?

With that you'd be able to still use a musl (cross) toolchain on a system that defaults on glibc and vice versa.

Reply:
Doesn't work for me.

```
~ armv7-alpine-linux-musleabihf-gcc -dumpversion
14.2.0
```

Reply:
`dumpversion` was a typo, I've amended the message with `-dumpmachine` a minute after, my bad.

```bash
~ # uname -a
Linux alpine 6.13.7-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Thu, 13 Mar 2025 18:11:42 +0000 x86_64 Linux
~ # gcc -dumpmachine
x86_64-alpine-linux-musl
```

Reply:
Cool, didn't know about that. I'll have to come back to this PR later, but this seems like a good improvement.

Reply:
I used `ifneq ($(findstring -musl,$(shell $(CROSS)gcc -dumpmachine)),)`. On my machine, the gcc output doesn't end with just "-musl", but with "-musleabihf".

Reply:
I see - I'd have probably written something like $filter(musl%, $lastword($(subst -, ,$(shell ...)))) to only check whether the last element of the machine triplet starts with "musl", but that's just me being pedantic :)

After all, unless you re-implement bits of cmake/autoconf in that makefile there's no proper way to find out if you're using a musl toolchain anyway, is there?